bindings/java/src/org/sleuthkit/datamodel/OsAccountAttribute.java

```diff
@@ -35,124 +35,118 @@
 	/**
 	 * Creates an os account attribute with int value.
 	 *
-	 * @param attributeType  Attribute type.
-	 * @param valueInt	      Int value.
-	 * @param osAccountObjId Obj id of account which the attribute pertains to. 
-	 * @param hostId         Id of host on which the attribute applies to. Pass
-	 *                       Null if it applies across hosts.
-	 * @param sourceObjId    Object id of the source where the attribute was
-	 *                       found.
+	 * @param attributeType	Attribute type.
+	 * @param valueInt	    Int value.
+	 * @param osAccount		Account which the attribute pertains to.
+	 * @param host			Host on which the attribute applies to. Pass Null if
+	 *                      it applies across hosts.
+	 * @param sourceObj		Source where the attribute was found.
 	 */
-	public OsAccountAttribute(BlackboardAttribute.Type attributeType, int valueInt, long osAccountObjId, Long hostId, Long sourceObjId) {
+	public OsAccountAttribute(BlackboardAttribute.Type attributeType, int valueInt, OsAccount osAccount, Host host, Content sourceObj) {
```

Comment:
In CT Standup today we discussed that 'sourceObj' for an account to be optional  - i.e caller could pass null for it. 

Do you think you can address it as part of this change? Otherwise I can address is separately after your PR is merged.

Reply:
I can take care of it.

Reply:
Thank you.  

In addition to the constructor you  changed I think this also needs:
 - change return value of getSourceObjId() from Long to Optional.
 - handle the null value when inserting in the database (OsAccountManager.addOsAccountAttributes(): line 678)